scripts/linux/setup.sh

```diff
@@ -0,0 +1,180 @@
+    popd
+}
+
+yes_or_exit() {
+        message=$1
+        while true; do
+                read -p "${message} [Y/n]" yn
+                case $yn in
+                        [Yy]* ) break;;
+                        [Nn]* ) exit;;
+                        * ) echo "Please answer yes or no.";;
+                esac
+        done
+}
+
+setup() {
+        while [[ $# -gt 0 ]]
+        do
+        key="$1"
+        case $key in
+                -y|--yes)
+                answer_yes="True"
+                shift # next argument
+                ;;
+                -h|--help)
+                display_help="True"
+                shift # next argument
+                ;;
+                -b|--builddir)
+                build_dir=$2
+                shift # past argument
+                shift # past value
+                ;;
+                -d|--depsdir)
+                deps_dir=$2
+                shift # past argument
+                shift # past value
+                ;;
+                -i|--depsinstalldir)
+                deps_install_dir=$2
+                shift # past argument
+                shift # past value
+                ;;
+                *)    # unknown option
+                POSITIONAL+=("$1") # save it in an array for later
+                shift # past argument
+                ;;
+        esac
+        done
+        set -- "${POSITIONAL[@]}" # restore positional parameters
+
+        if [[ "${display_help}" == "True" ]]; then
+                print_help
+                exit
+        fi
+        insta
+        ll_required_packages
+
+        source_dir=$(cd "$(dirname "$0")/../.."; pwd)
+
+        if [[ -z "${build_dir}" ]]; then
+                build_dir=$(pwd)/build
+        fi
+
+        echo "The sdk will be built in: ${build_dir}" ${quiet}
+
+        if [[ -z "${deps_dir}" ]]; then
+                deps_dir=$(pwd)/deps
+        fi
+
+        echo "The deps will be downloaded in: ${deps_dir}" ${quiet}
+
+        if [[ -z "${deps_install_dir}" ]]; then
+                deps_install_dir=${deps_dir}/installed
+        fi
+
+        echo "The deps will be installed in: ${deps_install_dir}" ${quiet}
+
+        if [[ -z "${answer_yes}" ]]; then
+             yes_or_exit "Do you want to continue?"
+        fi
+
+        mkdir -p "${build_dir}"
+        mkdir -p "${deps_dir}"
+        mkdir -p "${deps_install_dir}"
+
+        pushd "${deps_install_dir}"
+        deps_install_dir=$(pwd)
+        popd
+
+
+        get_deps_source_code ${deps_dir}
+
+        build_and_install_glog ${deps_dir}/glog ${deps_install_dir}/glog
+        build_and_install_protobuf ${deps_dir}/protobuf ${deps_install_dir}/protobuf
+        build_and_install_websockets ${deps_dir}/libwebsockets ${deps_install_dir}/websockets
+
+        pushd "${build_dir}"
+        cmake "${source_dir}" -DWITH_PYTHON=on -DWITH_OPENCV=on -DCMAKE_PREFIX_PATH="${deps_install_dir}/glog;${deps_install_dir}/protobuf;${deps_install_dir}/websockets"
+        make -j
```

Comment:
${NUM_JOBS}?

Reply:
I couldn't find a way to get the number of cores for the system. I think if it is not to much work an option like `-c x` for example that will set `NUM_JOBS=x` will work, or even using just `make` as it is right now will work. Indeed I forgot to remove the `-j`. Thanks.

Reply:
You can use `nproc --all` or use the -c option or both (use nproc by default and overwrite if user provides the core count)

Reply:
I have added a `-j|--jobs` option, as I see `-j` is used more often to specify the number of concurrent jobs. If that is not specified, the number of jobs will be the return value of `nproc --all`. Thanks for your suggestion.